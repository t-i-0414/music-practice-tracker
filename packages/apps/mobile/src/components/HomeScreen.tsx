import type React from 'react';
import { type PropsWithChildren } from 'react';

import { StyleSheet, Text, View } from 'react-native';

export const CustomText = ({ children }: PropsWithChildren): React.ReactNode => <Text>{children}</Text>;

const HomeScreen = (): React.ReactNode => (
  <View style={styles.container}>
    <CustomText>Welcome!</CustomText>
  </View>
);

export default HomeScreen;

const bgColor = '#fff'; // Default background color
const styles = StyleSheet.create({
  container: {
    alignItems: 'center',
    backgroundColor: bgColor,
    flex: 1,
    justifyContent: 'center',
  },
});
