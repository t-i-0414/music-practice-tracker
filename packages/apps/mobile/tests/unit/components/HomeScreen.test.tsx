import { render } from '@testing-library/react-native';

import HomeScreen, { CustomText } from '@/components/HomeScreen';

describe('<HomeScreen />', () => {
  it('text renders correctly on HomeScreen', () => {
    expect.hasAssertions();

    const { getByText } = render(<HomeScreen />);

    getByText('Welcome!');

    expect(1).toBe(1);
  });

  it('customText renders correctly', () => {
    expect.hasAssertions();

    const tree = render(<CustomText>Some text</CustomText>).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
