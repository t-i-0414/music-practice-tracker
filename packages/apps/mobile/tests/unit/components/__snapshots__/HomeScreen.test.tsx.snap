// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`<HomeScreen /> customText renders correctly 1`] = `
<Text>
  Some text
</Text>
`;
